drivers/tee/optee/rpc.c

```diff
@@ -237,52 +284,16 @@ static void handle_rpc_func_cmd_shm_alloc(struct tee_context *ctx,
 		return;
 	}
 
-	if (tee_shm_get_pa(shm, 0, &pa)) {
-		arg->ret = TEEC_ERROR_BAD_PARAMETERS;
+	rc = optee_rpc_process_shm_alloc(shm, &arg->params[0], &pages_list);
```

Comment:
Yes, I wrote it that way on purpose because `optee_rpc_process_shm_alloc` only uses the first element and not the others; I wanted to convey that meaning. But I'll change it to `arg->params` for simplicity's sake.